int32   velocity_left
int32   velocity_right
int32   distance_left
int32   distance_right
